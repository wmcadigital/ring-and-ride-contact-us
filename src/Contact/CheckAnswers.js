import { useFormState } from "react-final-form";
import PropTypes from "prop-types";

import EnquiryTypes from "./EnquiryTypes";
import Question from "../common/Question";
import Checkbox from "../common/Checkbox";
import FieldError from "../common/FieldError";
import Table from "../common/Table";
import CheckAnswerRow from "../common/CheckAnswerRow";
import CheckboxContainer from "../common/CheckboxContainer";

const CheckAnswers = ({ setGoToPage }) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;
  const enquiryLabel = EnquiryTypes[formValues.enquiryType]?.detailLabel;

  const agreeLegalError = stateApi.submitFailed ? stateApi.errors?.legal : null;

  return (
    <>
      <Question text="Check your answers" />
      <h3>About your {enquiryLabel} enquiry</h3>
      <Table>
        <CheckAnswerRow
          value={formValues["enquiryDetail"]}
          changeValueCallback={() => setGoToPage(1)}
        />
      </Table>
      <Table>
        <h3>About you</h3>
        <CheckAnswerRow
          label="Name"
          value={`${formValues["firstName"]} ${formValues["lastName"]}`}
          changeValueCallback={() => setGoToPage(2)}
        />
        <CheckAnswerRow
          label="Registration number"
          value={
            formValues["registrationNo"] ? formValues["registrationNo"] : ""
          }
          changeValueCallback={() => setGoToPage(3)}
        />
        <CheckAnswerRow
          label="How would you like to be contacted?"
          value={`${
            formValues["emailContact"]
              ? `Email: ${formValues["emailAddress"]} `
              : ""
          }
        ${formValues["phoneContact"] ? `Phone: ${formValues["phoneNo"]}` : ""}`}
          changeValueCallback={() => setGoToPage(4)}
        />
      </Table>
      <h3>Now send your enquiry</h3>
      <p>
        By submitting this enquiry you are confirming that, to the best of your
        knowledge, the details you are providing are correct.
      </p>
      <CheckboxContainer error={agreeLegalError}>
        <FieldError text={agreeLegalError} />
        <Checkbox
          fieldName="agreeTermsAndConditions"
          label={
            <>
              Agree to the{" "}
              <a
                href="https://www.wmca.org.uk/policies"
                target="_blank"
                rel="noreferrer"
              >
                terms and conditions
              </a>
            </>
          }
        />
        <Checkbox
          fieldName="agreePrivacyPolicy"
          label={
            <>
              Agree to the{" "}
              <a
                href="https://www.wmca.org.uk/policies"
                target="_blank"
                rel="noreferrer"
              >
                privacy policy
              </a>
            </>
          }
        />
      </CheckboxContainer>
    </>
  );
};

CheckAnswers.propTypes = {
  setGoToPage: PropTypes.func,
};

export default CheckAnswers;
