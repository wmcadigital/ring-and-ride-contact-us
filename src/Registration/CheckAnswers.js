import { useFormState } from "react-final-form";
import PropTypes from "prop-types";

import Question from "../common/Question";
import Checkbox from "../common/Checkbox";
import FieldError from "../common/FieldError";
import Table from "../common/Table";
import CheckAnswerRow from "../common/CheckAnswerRow";
import CheckboxContainer from "../common/CheckboxContainer";
import formatDate from "../common/formatDate";
import formatOptions from "../common/formatOptions";
import AddressFormatted from "../common/AddressFormatted";

import contactPreferenceLabels from "./section1/ContactPreferenceOptions";
import Ethnicity, { PreferNotToSayLabel } from "./section1/Ethnicity";
import { BusProblemOptions } from "./section3/StandardBusProblems";
import { ReasonOptions } from "./section3/StandardBusReasons";
import { MobilityAidsOptions } from "./section3/MobilityAidsQuery";
import getAboutSectionName from "./section1/getAboutSectionName";
import getRequirementsSectionName from "./section3/getRequirementsSectionName";

const CheckAnswers = ({ setGoToPage }) => {
  const stateApi = useFormState();
  const formValues = stateApi.values;

  const agreeLegalError = stateApi.submitFailed ? stateApi.errors?.legal : null;

  const registerForYourself = formValues["registerForYourself"] === "yes";

  const ethnicity = formValues["ethnicity"];
  const specificEthnicity = formValues["specificEthnicity"];
  const ethnicityNotGiven = ethnicity === "preferNotToSay";
  const specificEthnicityNotGiven = specificEthnicity === "preferNotToSay";
  const preferNotToSayLabel = PreferNotToSayLabel["preferNotToSay"];
  const ethnicityDisplay = ethnicityNotGiven
    ? preferNotToSayLabel
    : Ethnicity[ethnicity].label;
  const specificEthnicityDisplay = ethnicityNotGiven
    ? ""
    : ` - ${
        specificEthnicityNotGiven
          ? preferNotToSayLabel
          : Ethnicity[ethnicity].specific[specificEthnicity].label
      }`;

  const baseOffSet = ethnicityNotGiven ? 7 : 8;
  const offSet1 =
    !registerForYourself && formValues["emergencyContact"] === "no"
      ? baseOffSet + 1
      : baseOffSet;

  const emergencyContactPresent =
    formValues["emergencyContact"] === "yes" ||
    formValues["emergencyContactAnother"] === "yes";

  const offSet2 = emergencyContactPresent ? offSet1 + 6 : offSet1 + 1;

  const offSet3 =
    formValues["hasCondition"] == "yes" ? offSet2 + 4 : offSet2 + 3;

  return (
    <>
      <Question text="Check your answers" />
      <h3>{getAboutSectionName(stateApi)}</h3>
      <Table>
        <CheckAnswerRow
          label="Name"
          value={`${formValues["firstName"]} ${formValues["lastName"]}`}
          changeValueCallback={() => setGoToPage(1)}
        />
        <CheckAnswerRow
          label="Date of birth"
          value={formatDate(
            formValues["bdayDay"],
            formValues["bdayMonth"],
            formValues["bdayYear"]
          )}
          changeValueCallback={() => setGoToPage(2)}
        />
        <CheckAnswerRow
          label="Telephone number"
          value={formValues["phoneNo"]}
          changeValueCallback={() => setGoToPage(3)}
        />
        <CheckAnswerRow
          label="Email address"
          value={formValues["emailAddress"]}
          changeValueCallback={() => setGoToPage(4)}
        />
        <CheckAnswerRow
          label="Address"
          value={<AddressFormatted addressObj={formValues["registered"]} />}
          changeValueCallback={() => setGoToPage(5)}
        />
        <CheckAnswerRow
          label="Preferred method of contact"
          value={formatOptions(
            formValues["contactPreference"],
            contactPreferenceLabels
          )}
          changeValueCallback={() => setGoToPage(6)}
        />
        <CheckAnswerRow
          label="Ethnicity"
          value={`${ethnicityDisplay}${specificEthnicityDisplay}`}
          changeValueCallback={() => setGoToPage(7)}
        />
      </Table>
      {emergencyContactPresent ? (
        <>
          <h3>Emergency Contact</h3>
          <Table>
            <CheckAnswerRow
              label="Name"
              value={`${formValues["emergencyFirstName"]} ${formValues["emergencyLastName"]}`}
              changeValueCallback={() => setGoToPage(offSet1 + 2)}
            />
            <CheckAnswerRow
              label={
                registerForYourself
                  ? "Relationship to you"
                  : "Relationship to applicant"
              }
              value={formValues["emergencyRelationship"]}
              changeValueCallback={() => setGoToPage(offSet1 + 3)}
            />
            <CheckAnswerRow
              label="Telephone number"
              value={formValues["emergencyPhoneNo"]}
              changeValueCallback={() => setGoToPage(offSet1 + 4)}
            />
            <CheckAnswerRow
              label="Email address"
              value={formValues["emergencyEmailAddress"]}
              changeValueCallback={() => setGoToPage(offSet1 + 5)}
            />
            <CheckAnswerRow
              label="Address"
              value={<AddressFormatted addressObj={formValues["emergency"]} />}
              changeValueCallback={() => setGoToPage(offSet1 + 6)}
            />
          </Table>
        </>
      ) : null}
      <h3>{getRequirementsSectionName(stateApi)}</h3>
      <Table>
        <CheckAnswerRow
          label="Problems accessing standard services"
          value={formatOptions(
            formValues["standardBusProb"],
            BusProblemOptions(registerForYourself)
          )}
          changeValueCallback={() => setGoToPage(offSet2 + 1)}
        />
        <CheckAnswerRow
          label="Reasons"
          value={`${formatOptions(
            formValues["standardBusReason"],
            ReasonOptions(registerForYourself)
          )}${
            formValues["standardBusReason"]?.other === true
              ? `, ${formValues["standardBusReason"].otherReason}`
              : ""
          }`}
          changeValueCallback={() => setGoToPage(offSet2 + 2)}
        />
        {formValues["hasCondition"] == "yes" ? (
          <CheckAnswerRow
            label={
              registerForYourself
                ? "What conditions do you have?"
                : "What conditions do they have?"
            }
            value={formValues["conditionDetail"]}
            changeValueCallback={() => setGoToPage(offSet2 + 4)}
          />
        ) : null}
        <CheckAnswerRow
          label="Use of a mobility aid"
          value={`${formatOptions(
            formValues["mobilityAids"],
            MobilityAidsOptions(registerForYourself)
          )}${
            formValues["mobilityAids"]?.other === true
              ? `, ${formValues["mobilityAids"].otherRequirement}`
              : ""
          }`}
          changeValueCallback={() => setGoToPage(offSet3 + 1)}
        />
        {formValues["mobilityAids"]?.manualWheelchair ||
        formValues["mobilityAids"]?.poweredWheelchair ? (
          <CheckAnswerRow
            label={
              formValues["mobilityAids"]?.manualWheelchair
                ? "Manual wheelchair"
                : "Powered wheelchair"
            }
            value={`${registerForYourself ? "I " : "They "}${
              formValues["wheelChairTransfer"] === "yes"
                ? "can transfer"
                : "cannot transfer"
            }`}
            changeValueCallback={() => setGoToPage(offSet3 + 2)}
          />
        ) : null}
        {formValues["additionalRequirements"] == "yes" ? (
          <CheckAnswerRow
            label="Additional requirements"
            value={formValues["additionalRequirementsDetails"]}
            changeValueCallback={() => setGoToPage(offSet3 + 3)}
          />
        ) : null}
      </Table>
      <h3>Now send your request</h3>
      <p>
        By submitting this enquiry you are confirming that, to the best of your
        knowledge, the details you are providing are correct.
      </p>
      <CheckboxContainer error={agreeLegalError}>
        <FieldError text={agreeLegalError} />
        <Checkbox
          fieldName="agreeRingAndRide"
          label={
            <span>
              Agree that you need to use Ring & Ride because you find it either
              impossible or difficult to use other forms of public transport.
              You will inform Ring & Ride if there is any change in your
              circumstances that affects your eligibility to use the service.
            </span>
          }
        />
        <Checkbox
          fieldName="agreeTermsAndConditions"
          label={
            <>
              Agree to the{" "}
              <a
                href="https://www.wmca.org.uk/policies"
                target="_blank"
                rel="noreferrer"
              >
                terms and conditions
              </a>
            </>
          }
        />
        <Checkbox
          fieldName="agreePrivacyPolicy"
          label={
            <>
              Agree to the{" "}
              <a
                href="https://www.wmca.org.uk/policies"
                target="_blank"
                rel="noreferrer"
              >
                privacy policy
              </a>
            </>
          }
        />
      </CheckboxContainer>
    </>
  );
};

CheckAnswers.propTypes = {
  setGoToPage: PropTypes.func,
};

export default CheckAnswers;
