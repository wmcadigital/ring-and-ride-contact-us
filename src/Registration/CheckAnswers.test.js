import { useFormState } from "react-final-form";
import { createRenderer } from "react-test-renderer/shallow";

import CheckAnswers from "./CheckAnswers";

jest.mock("react-final-form");

const formApiReturn = {
  values: {
    registerForYourself: "yes",
    firstName: "Chan",
    lastName: "Singh",
    bdayDay: "01",
    bdayMonth: "01",
    bdayYear: "1990",
    phoneNo: "87098908",
    emailAddress: "test@testerson.com",
    registered: {
      searchPostCode: "B19 2NN",
      addressId: "f2c8b89b-4b11-4557-93cb-9f846748bcba",
      addressLine1: "Church Of Christ",
      addressLine2: "Summer Lane",
      townOrCity: "Birmingham",
      county: "West Midlands",
      postCode: "B19 2NN",
    },
    contactPreference: {
      email: true,
      letterAndEmail: false,
      largePrintLetter: true,
    },
    ethnicity: "asian",
    specificEthnicity: "indian",
    emergencyContact: "yes",
    emergencyFirstName: "Test",
    emergencyLastName: "Testerson",
    emergencyRelationship: "Carer",
    emergencyPhoneNo: "0800 555500",
    emergencyEmailAddress: "tester@testerson.com",
    emergency: {
      searchPostCode: "B19 3SD",
      addressId: "46091f33-ae00-4709-bbc8-38145f2233b1",
      addressLine1: "West Midlands Combined Authority",
      addressLine2: "16 Summer Lane",
      townOrCity: "Birmingham",
      county: "West Midlands",
      postCode: "B19 3SD",
    },
    standardBusProb: {
      cannotGetToBus: true,
      cannotTravelOnBus: true,
    },
    standardBusReason: {
      limitedMobility: true,
      other: true,
      otherReason: "Arthiritis",
      companionOrCarer: true,
    },
    hasCondition: "yes",
    conditionDetail: "Arthiritis",
    mobilityAids: {
      manualWheelchair: true,
      walkingFrame: true,
    },
    wheelChairTransfer: "no",
    additionalRequirements: "yes",
    additionalRequirementsDetails: "Extra requirement",
  },
};

describe("CheckAnswers", () => {
  it("renders CheckAnswers correctly", () => {
    useFormState.mockImplementationOnce(() => formApiReturn);
    const renderer = createRenderer();

    renderer.render(<CheckAnswers />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders CheckAnswers - register on behalf of someone else", () => {
    const formApiReturnOnBehalf = {
      ...formApiReturn,
      values: { ...formApiReturn.values, registerForYourself: "no" },
    };
    useFormState.mockImplementationOnce(() => formApiReturnOnBehalf);
    const renderer = createRenderer();

    renderer.render(<CheckAnswers />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders CheckAnswers - no emergency", () => {
    const formApiReturnNoEmergency = {
      ...formApiReturn,
      values: { ...formApiReturn.values, emergencyContact: "no" },
    };
    useFormState.mockImplementationOnce(() => formApiReturnNoEmergency);
    const renderer = createRenderer();

    renderer.render(<CheckAnswers />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });

  it("renders CheckAnswers with an error if present", () => {
    const formApiReturnWithError = {
      ...formApiReturn,
      submitFailed: true,
      errors: { legal: "Required" },
    };
    useFormState.mockImplementationOnce(() => formApiReturnWithError);
    const renderer = createRenderer();

    renderer.render(<CheckAnswers />);

    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
