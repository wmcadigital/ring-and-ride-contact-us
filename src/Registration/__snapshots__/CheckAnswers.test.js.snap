// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckAnswers renders CheckAnswers - no emergency 1`] = `
<React.Fragment>
  <Question
    text="Check your answers"
  />
  <h3>
    About you
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Name"
      value="Chan Singh"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Date of birth"
      value="1 January 1990"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Telephone number"
      value="87098908"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Email address"
      value="test@testerson.com"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Address"
      value={
        <AddressFormatted
          addressObj={
            Object {
              "addressId": "f2c8b89b-4b11-4557-93cb-9f846748bcba",
              "addressLine1": "Church Of Christ",
              "addressLine2": "Summer Lane",
              "county": "West Midlands",
              "postCode": "B19 2NN",
              "searchPostCode": "B19 2NN",
              "townOrCity": "Birmingham",
            }
          }
        />
      }
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Preferred method of contact"
      value="Email, Large print letter"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Ethnicity"
      value="Asian or Asian British - Indian"
    />
  </Table>
  <h3>
    Your requirements
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Problems accessing standard services"
      value="I cannot get to a bus stop, I cannot travel on a bus"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Reasons"
      value="I have limited mobility, I cannot travel without a companion or carer, Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="What conditions do you have?"
      value="Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Use of a mobility aid"
      value="I use a manual wheelchair, I use a walking frame"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Manual wheelchair"
      value="I cannot transfer"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Additional requirements"
      value="Extra requirement"
    />
  </Table>
  <h3>
    Now send your request
  </h3>
  <p>
    By submitting this enquiry you are confirming that, to the best of your knowledge, the details you are providing are correct.
  </p>
  <CheckboxContainer
    error={null}
  >
    <FieldError
      text={null}
    />
    <Checkbox
      fieldName="agreeRingAndRide"
      label={
        <span>
          Agree that you need to use Ring & Ride because you find it either impossible or difficult to use other forms of public transport. You will inform Ring & Ride if there is any change in your circumstances that affects your eligibility to use the service.
        </span>
      }
    />
    <Checkbox
      fieldName="agreeTermsAndConditions"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            terms and conditions
          </a>
        </React.Fragment>
      }
    />
    <Checkbox
      fieldName="agreePrivacyPolicy"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            privacy policy
          </a>
        </React.Fragment>
      }
    />
  </CheckboxContainer>
</React.Fragment>
`;

exports[`CheckAnswers renders CheckAnswers - register on behalf of someone else 1`] = `
<React.Fragment>
  <Question
    text="Check your answers"
  />
  <h3>
    About the applicant
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Name"
      value="Chan Singh"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Date of birth"
      value="1 January 1990"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Telephone number"
      value="87098908"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Email address"
      value="test@testerson.com"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Address"
      value={
        <AddressFormatted
          addressObj={
            Object {
              "addressId": "f2c8b89b-4b11-4557-93cb-9f846748bcba",
              "addressLine1": "Church Of Christ",
              "addressLine2": "Summer Lane",
              "county": "West Midlands",
              "postCode": "B19 2NN",
              "searchPostCode": "B19 2NN",
              "townOrCity": "Birmingham",
            }
          }
        />
      }
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Preferred method of contact"
      value="Email, Large print letter"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Ethnicity"
      value="Asian or Asian British - Indian"
    />
  </Table>
  <React.Fragment>
    <h3>
      Emergency Contact
    </h3>
    <Table>
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Name"
        value="Test Testerson"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Relationship to applicant"
        value="Carer"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Telephone number"
        value="0800 555500"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Email address"
        value="tester@testerson.com"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Address"
        value={
          <AddressFormatted
            addressObj={
              Object {
                "addressId": "46091f33-ae00-4709-bbc8-38145f2233b1",
                "addressLine1": "West Midlands Combined Authority",
                "addressLine2": "16 Summer Lane",
                "county": "West Midlands",
                "postCode": "B19 3SD",
                "searchPostCode": "B19 3SD",
                "townOrCity": "Birmingham",
              }
            }
          />
        }
      />
    </Table>
  </React.Fragment>
  <h3>
    Applicant requirements
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Problems accessing standard services"
      value="They cannot get to a bus stop, They cannot travel on a bus"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Reasons"
      value="They have limited mobility, They cannot travel without a companion or carer, Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="What conditions do they have?"
      value="Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Use of a mobility aid"
      value="They use a manual wheelchair, They use a walking frame"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Manual wheelchair"
      value="They cannot transfer"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Additional requirements"
      value="Extra requirement"
    />
  </Table>
  <h3>
    Now send your request
  </h3>
  <p>
    By submitting this enquiry you are confirming that, to the best of your knowledge, the details you are providing are correct.
  </p>
  <CheckboxContainer
    error={null}
  >
    <FieldError
      text={null}
    />
    <Checkbox
      fieldName="agreeRingAndRide"
      label={
        <span>
          Agree that you need to use Ring & Ride because you find it either impossible or difficult to use other forms of public transport. You will inform Ring & Ride if there is any change in your circumstances that affects your eligibility to use the service.
        </span>
      }
    />
    <Checkbox
      fieldName="agreeTermsAndConditions"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            terms and conditions
          </a>
        </React.Fragment>
      }
    />
    <Checkbox
      fieldName="agreePrivacyPolicy"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            privacy policy
          </a>
        </React.Fragment>
      }
    />
  </CheckboxContainer>
</React.Fragment>
`;

exports[`CheckAnswers renders CheckAnswers correctly 1`] = `
<React.Fragment>
  <Question
    text="Check your answers"
  />
  <h3>
    About you
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Name"
      value="Chan Singh"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Date of birth"
      value="1 January 1990"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Telephone number"
      value="87098908"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Email address"
      value="test@testerson.com"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Address"
      value={
        <AddressFormatted
          addressObj={
            Object {
              "addressId": "f2c8b89b-4b11-4557-93cb-9f846748bcba",
              "addressLine1": "Church Of Christ",
              "addressLine2": "Summer Lane",
              "county": "West Midlands",
              "postCode": "B19 2NN",
              "searchPostCode": "B19 2NN",
              "townOrCity": "Birmingham",
            }
          }
        />
      }
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Preferred method of contact"
      value="Email, Large print letter"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Ethnicity"
      value="Asian or Asian British - Indian"
    />
  </Table>
  <React.Fragment>
    <h3>
      Emergency Contact
    </h3>
    <Table>
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Name"
        value="Test Testerson"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Relationship to you"
        value="Carer"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Telephone number"
        value="0800 555500"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Email address"
        value="tester@testerson.com"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Address"
        value={
          <AddressFormatted
            addressObj={
              Object {
                "addressId": "46091f33-ae00-4709-bbc8-38145f2233b1",
                "addressLine1": "West Midlands Combined Authority",
                "addressLine2": "16 Summer Lane",
                "county": "West Midlands",
                "postCode": "B19 3SD",
                "searchPostCode": "B19 3SD",
                "townOrCity": "Birmingham",
              }
            }
          />
        }
      />
    </Table>
  </React.Fragment>
  <h3>
    Your requirements
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Problems accessing standard services"
      value="I cannot get to a bus stop, I cannot travel on a bus"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Reasons"
      value="I have limited mobility, I cannot travel without a companion or carer, Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="What conditions do you have?"
      value="Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Use of a mobility aid"
      value="I use a manual wheelchair, I use a walking frame"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Manual wheelchair"
      value="I cannot transfer"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Additional requirements"
      value="Extra requirement"
    />
  </Table>
  <h3>
    Now send your request
  </h3>
  <p>
    By submitting this enquiry you are confirming that, to the best of your knowledge, the details you are providing are correct.
  </p>
  <CheckboxContainer
    error={null}
  >
    <FieldError
      text={null}
    />
    <Checkbox
      fieldName="agreeRingAndRide"
      label={
        <span>
          Agree that you need to use Ring & Ride because you find it either impossible or difficult to use other forms of public transport. You will inform Ring & Ride if there is any change in your circumstances that affects your eligibility to use the service.
        </span>
      }
    />
    <Checkbox
      fieldName="agreeTermsAndConditions"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            terms and conditions
          </a>
        </React.Fragment>
      }
    />
    <Checkbox
      fieldName="agreePrivacyPolicy"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            privacy policy
          </a>
        </React.Fragment>
      }
    />
  </CheckboxContainer>
</React.Fragment>
`;

exports[`CheckAnswers renders CheckAnswers with an error if present 1`] = `
<React.Fragment>
  <Question
    text="Check your answers"
  />
  <h3>
    About you
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Name"
      value="Chan Singh"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Date of birth"
      value="1 January 1990"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Telephone number"
      value="87098908"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Email address"
      value="test@testerson.com"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Address"
      value={
        <AddressFormatted
          addressObj={
            Object {
              "addressId": "f2c8b89b-4b11-4557-93cb-9f846748bcba",
              "addressLine1": "Church Of Christ",
              "addressLine2": "Summer Lane",
              "county": "West Midlands",
              "postCode": "B19 2NN",
              "searchPostCode": "B19 2NN",
              "townOrCity": "Birmingham",
            }
          }
        />
      }
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Preferred method of contact"
      value="Email, Large print letter"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Ethnicity"
      value="Asian or Asian British - Indian"
    />
  </Table>
  <React.Fragment>
    <h3>
      Emergency Contact
    </h3>
    <Table>
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Name"
        value="Test Testerson"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Relationship to you"
        value="Carer"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Telephone number"
        value="0800 555500"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Email address"
        value="tester@testerson.com"
      />
      <CheckAnswerRow
        changeValueCallback={[Function]}
        label="Address"
        value={
          <AddressFormatted
            addressObj={
              Object {
                "addressId": "46091f33-ae00-4709-bbc8-38145f2233b1",
                "addressLine1": "West Midlands Combined Authority",
                "addressLine2": "16 Summer Lane",
                "county": "West Midlands",
                "postCode": "B19 3SD",
                "searchPostCode": "B19 3SD",
                "townOrCity": "Birmingham",
              }
            }
          />
        }
      />
    </Table>
  </React.Fragment>
  <h3>
    Your requirements
  </h3>
  <Table>
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Problems accessing standard services"
      value="I cannot get to a bus stop, I cannot travel on a bus"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Reasons"
      value="I have limited mobility, I cannot travel without a companion or carer, Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="What conditions do you have?"
      value="Arthiritis"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Use of a mobility aid"
      value="I use a manual wheelchair, I use a walking frame"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Manual wheelchair"
      value="I cannot transfer"
    />
    <CheckAnswerRow
      changeValueCallback={[Function]}
      label="Additional requirements"
      value="Extra requirement"
    />
  </Table>
  <h3>
    Now send your request
  </h3>
  <p>
    By submitting this enquiry you are confirming that, to the best of your knowledge, the details you are providing are correct.
  </p>
  <CheckboxContainer
    error="Required"
  >
    <FieldError
      text="Required"
    />
    <Checkbox
      fieldName="agreeRingAndRide"
      label={
        <span>
          Agree that you need to use Ring & Ride because you find it either impossible or difficult to use other forms of public transport. You will inform Ring & Ride if there is any change in your circumstances that affects your eligibility to use the service.
        </span>
      }
    />
    <Checkbox
      fieldName="agreeTermsAndConditions"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            terms and conditions
          </a>
        </React.Fragment>
      }
    />
    <Checkbox
      fieldName="agreePrivacyPolicy"
      label={
        <React.Fragment>
          Agree to the
           
          <a
            href="https://www.wmca.org.uk/policies"
            rel="noreferrer"
            target="_blank"
          >
            privacy policy
          </a>
        </React.Fragment>
      }
    />
  </CheckboxContainer>
</React.Fragment>
`;
